src/common/dao/project/projectmember_test.go

```diff
@@ -325,6 +327,21 @@ func TestGetTotalOfProjectMembers(t *testing.T) {
 	}
 }
 
+func TestListRoles(t *testing.T) {
+	roles, err := ListRoles(nil, 1)
+	require.Nil(t, err)
+	assert.Len(t, roles, 0)
+
+	user, err := dao.GetUser(models.User{Username: "member_test_01"})
+	require.Nil(t, err)
+	project, err := dao.GetProjectByName("member_test_01")
+	require.Nil(t, err)
+
+	roles, err = ListRoles(user, project.ProjectID)
```

Comment:
Does it cover the case that user in groups?

Reply:
Case updated to cover the groups